src/generate.js

```diff
@@ -24,45 +24,42 @@ module.exports = function generate(parameters) {
     adaptiveIconManifests: [],
   };
 
-  return findIconsetFolders(searchRoot)
-    .then(iconSets => Promise.all(iconSets.map((iconset) => {
-      if (!platforms.includes('ios')) return null;
+  const iconSets = await findIconsetFolders(searchRoot);
+  await Promise.all(iconSets.map(async (iconset) => {
+    if (!platforms.includes('ios')) return null;
+    console.log(`Found iOS iconset: ${iconset}...`);
 
-      console.log(`Found iOS iconset: ${iconset}...`);
+    const { icons } = await generateIconsetIcons(sourceIcon, iconset);
+    results.iconsets.push({ iconset, icons });
+    icons.forEach((icon) => {
+      console.log(`    ${chalk.green('✓')}  Generated icon ${icon}`);
+    });
+    console.log(`    ${chalk.green('✓')}  Updated Contents.json`);
 
-      return generateIconsetIcons(sourceIcon, iconset)
-        .then(({ icons }) => {
-          results.iconsets.push({ iconset, icons });
-          icons.forEach((icon) => {
-            console.log(`    ${chalk.green('✓')}  Generated icon ${icon}`);
-          });
-          console.log(`    ${chalk.green('✓')}  Updated Contents.json`);
+    return null;
+  }));
+  const manifests = await findAndroidManifests(searchRoot);
+  await Promise.all(manifests.map(async (manifest) => {
+    if (!platforms.includes('android')) return null;
+    console.log(`Found Android Manifest: ${manifest}...`);
+    const operations = [
```

Comment:
Another place we can simplify logic I think

Reply:
Ok - let me look at all the `operations` code smells as part of this PR